src/puppetlabs/trapperkeeper/services/webserver/jetty9_core.clj

```diff
@@ -270,25 +273,50 @@
   (connector* server config nil))
 
 (schema/defn ^:always-validate
-  queue-thread-pool :- QueuedThreadPool
-  [max-threads :- schema/Int
-   queue-max-size :- schema/Int]
-  (let [queue-min-size (min config/default-queue-min-threads queue-max-size)]
-    (QueuedThreadPool. max-threads
-                       queue-min-size
-                       config/default-queue-idle-timeout
-                       (BlockingArrayQueue.
-                         queue-min-size
-                         queue-min-size
-                         queue-max-size))))
+  queue-thread-pool :- (schema/maybe QueuedThreadPool)
+  [max-threads :- (schema/maybe schema/Int)
+   queue-max-size :- (schema/maybe schema/Int)]
+  (if (or max-threads queue-max-size)
+    (let [thread-pool (if max-threads
+                        (QueuedThreadPool. max-threads)
+                        (QueuedThreadPool.))]
+      (if queue-max-size
+        ;; The code below is definitely not ideal, but there isn't a way to set
+        ;; the maximum capacity of the QueuedThreadPool's BlockingArrayQueue
+        ;; after construction.  We're trying to avoid hard-coding our own
+        ;; defaults for other settings that we want Jetty to control, e.g., the
+        ;; initial capacity of the queue and minimum number of threads.  By
+        ;; reconstructing the QueuedThreadPool here, we can use Jetty's defaults
+        ;; for settings unrelated to `queue-max-size`.  QueuedThreadPool and
+        ;; BlockingArrayQueue construction isn't too expensive.  It mostly
+        ;; involves some initial memory allocations.  The more expensive work -
+        ;; where threads are actually started and the queue expands to fulfill
+        ;; new requests - would only happen after the QueuedThreadPool were
+        ;; started.  That won't happen for the `thread-pool` instance from
+        ;; above, which just gets thrown away as this function falls out of
+        ;; scope without ever having been started.  Also, this function would
+        ;; likely only be called once per server startup where a
+        ;; `queue-max-size` were configured.
+        (let [min-threads    (.getMinThreads thread-pool)
+              queue-min-size (min queue-max-size min-threads)]
+          (QueuedThreadPool. (.getMaxThreads thread-pool)
+                             min-threads
+                             (.getIdleTimeout thread-pool)
+                             (BlockingArrayQueue.
+                               queue-min-size
+                               queue-min-size
+                               queue-max-size)))
```

Comment:
Oh, my bad, I was mistaken on what those 2 constructor params were.

I think that change would be an improvement, but I'm fine without it too - however, a comment would still be really helpful.

And thanks for talking me through this.